commandapi-platforms/commandapi-paper/commandapi-paper-core/src/main/java/dev/jorel/commandapi/CommandAPIPaper.java

```diff
@@ -0,0 +1,308 @@
+	}
+
+	@Override
+	public BukkitCommandSender<? extends CommandSender> wrapCommandSender(CommandSender sender) {
+		if (sender instanceof BlockCommandSender block) {
+			return new BukkitBlockCommandSender(block);
+		}
+		if (sender instanceof ConsoleCommandSender console) {
+			return new BukkitConsoleCommandSender(console);
+		}
+		if (sender instanceof Player player) {
+			return new BukkitPlayer(player);
+		}
+		if (sender instanceof org.bukkit.entity.Entity entity) {
+			return new BukkitEntity(entity);
+		}
+		if (sender instanceof NativeProxyCommandSender nativeProxy) {
+			return new BukkitNativeProxyCommandSender(nativeProxy);
+		}
+		if (sender instanceof ProxiedCommandSender proxy) {
+			return new BukkitProxiedCommandSender(proxy);
+		}
+		if (sender instanceof RemoteConsoleCommandSender remote) {
+			return new BukkitRemoteConsoleCommandSender(remote);
+		}
+		if (this.feedbackForwardingCommandSender.isInstance(sender)) {
+			// We literally cannot type this at compile-time, so let's use a placeholder CommandSender instance
+			return new BukkitFeedbackForwardingCommandSender<CommandSender>(this.feedbackForwardingCommandSender.cast(sender));
+		}
+		if (this.nullCommandSender != null && this.nullCommandSender.isInstance(sender)) {
+			// Since this should only be during a function load or setting up
+			//  help topics, this is just a placeholder to evade the exception.
+			return null;
+		}
+		throw new RuntimeException("Failed to wrap CommandSender " + sender + " to a CommandAPI-compatible BukkitCommandSender");
+	}
+
+	/**
+	 * Forces a command to return a success value of 0
+	 *
+	 * @param message Description of the error message, formatted as an adventure chat component
+	 * @return a {@link WrapperCommandSyntaxException} that wraps Brigadier's
+	 * {@link com.mojang.brigadier.exceptions.CommandSyntaxException}
+	 */
+	public static WrapperCommandSyntaxException failWithAdventureComponent(ComponentLike message) {
+		return CommandAPI.failWithMessage(BukkitTooltip.messageFromAdventureComponent(message));
+	}
+
+	@SuppressWarnings("ConstantValue")
+	@Override
+	public CommandAPILogger getLogger() {
+		if (Bukkit.getServer() != null) {
+			return super.getLogger();
+		}
+		if (bootstrapLogger == null) {
+			bootstrapLogger = new BootstrapLogger();
+		}
+		return bootstrapLogger;
+	}
+
+	private static class BootstrapLogger implements CommandAPILogger {
+		private final ComponentLogger componentLogger;
+
+		protected BootstrapLogger() {
+			this.componentLogger = ComponentLogger.logger("CommandAPI");
+		}
+
+		@Override
+		public void info(String message) {
+			componentLogger.info(message);
+		}
+
+		@Override
+		public void warning(String message) {
+			componentLogger.warn(message);
+		}
+
+		@Override
+		public void severe(String message) {
+			componentLogger.error(message);
+		}
+
+		@Override
+		public void severe(String message, Throwable exception) {
+			componentLogger.error(message, exception);
+		}
+	}
+
```

Comment:
There is a utility method for turning a slf4j logger (which `ComponentLogger` extends) into a `CommandAPILogger`.

```suggestion
	@SuppressWarnings("ConstantValue")
	@Override
	public CommandAPILogger getLogger() {
		if (Bukkit.getServer() != null) {
			return super.getLogger();
		}
		if (bootstrapLogger == null) {
			bootstrapLogger = CommandAPILogger.fromSlf4jLogger(ComponentLogger.logger("CommandAPI"));
		}
		return bootstrapLogger;
	}
```